src/4_dbscan.md

```diff
@@ -0,0 +1,51 @@
+## Using DBSCAN with `linfa-clustering`
+
+### What is the DBSCAN algorithm?
+
+The DBSCAN algorithm (Density-Based Spatial Clustering Algorithm with Noise) was originally published in [1996](https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.121.9220), and has since become one of the most popular and well-known clustering algorithms available. 
+
+
+### Comparison with KMeans
+
+Before getting into the code, let's examine how these differences in approach plays out on different types of data. In the images below, both the DBSCAN and KMeans algorithms were applied to the same dataset. The KMeans algorithm was manually set to find 3 clusters (remember, DBSCAN automatically calculates the number of clusters based on the provided parameters).
+
+<img src="clustering_comparison.png" alt="Comparison" width=600px height=300px align="middle">
+
+This example[^1] demonstrates two of the major strengths of DBSCAN over an algorithm like KMeans; it is able to automatically detect the number of clusters that meet the set of given parameters. Keep in mind that this doesn't mean DBSCAN require less information about the dataset, but rather that the information it does require differs from an algorithm like KMeans.
+
+DBSCAN does a great job at finding clustering that are spatially contiguous, but not necessarily confined to single region. This is where the "and Noise" part of the algorithm's name comes in. Especially in real-world data, there's often data that won't fit well into a given cluster. These can be outliers or points that don't demonstrate good alignment with any of the main clusters. DBSCAN doesn't require that they do. Instead, it will simply give them a cluster label of `None` (in our example, these are graphically the black points). 
+
+Comparatively, KMeans will take into account each point in the dataset, which means outliers can negatively affect the local optimal location for a given cluster's centroid in order to accommodate them. 
```

Comment:
Also `Euclidean space is linear, which means that small changes in the data result in proportionately small changes to the position of the centroids. This is problematic when there are outliers in the data.`